src/main/java/dev/waterdog/player/HandshakeUtils.java

```diff
@@ -130,16 +130,19 @@ public static HandshakeEntry processHandshake(BedrockSession session, LoginPacke
         ECPublicKey identityPublicKey = EncryptionUtils.generateKey(identityPublicKeyString);
         JWSObject clientJwt = JWSObject.parse(packet.getSkinData().toString());
         EncryptionUtils.verifyJwt(clientJwt, identityPublicKey);
-        JsonObject clientData = HandshakeUtils.parseClientData(clientJwt, session);
+        JsonObject clientData = HandshakeUtils.parseClientData(clientJwt, extraData, session);
         return new HandshakeEntry(identityPublicKey, clientData, extraData, xboxAuth, protocol);
     }
 
-    public static JsonObject parseClientData(JWSObject clientJwt, BedrockSession session) throws Exception {
+    public static JsonObject parseClientData(JWSObject clientJwt,JsonObject extraData, BedrockSession session) throws Exception {
```

Comment:
Forgot the space after `,` here. Please add it 😄

Reply:
OK